src/lib/updateElement.js

```diff
@@ -1,6 +1,106 @@
 import { addEvent, removeEvent } from "./eventManager";
 import { createElement } from "./createElement.js";
+import { isClassName, isEvent } from "./util.js";
 
-function updateAttributes(target, originNewProps, originOldProps) {}
+function isChangedAttributes(originNewProps, originOldProps) {
+  if (
+    (!originNewProps && originOldProps) ||
+    (originNewProps && !originOldProps)
+  ) {
+    return true;
+  }
 
-export function updateElement(parentElement, newNode, oldNode, index = 0) {}
+  const mergedProps = { ...originOldProps, ...originNewProps };
+  return Object.keys(mergedProps ?? {}).some(
+    (key) => mergedProps[key] !== originOldProps[key],
+  );
+}
+
+function updateAttributes(target, originNewProps, originOldProps) {
+  Object.keys(originOldProps ?? {}).forEach((key) => {
+    if (isEvent(key)) {
+      const eventType = key.slice(2).toLowerCase();
+      removeEvent(target, eventType);
+      return;
+    }
+
+    if (isClassName(key)) {
+      target.removeAttribute("class");
+      return;
+    }
+
+    target.removeAttribute(key);
+  });
+
+  Object.keys(originNewProps ?? {}).forEach((key) => {
+    if (isEvent(key)) {
+      const eventType = key.slice(2).toLowerCase();
+      addEvent(target, eventType, originNewProps[key]);
+      return;
+    }
+
+    if (isClassName(key)) {
+      target.setAttribute("class", originNewProps[key]);
+      return;
+    }
+
+    target.setAttribute(key, originNewProps[key]);
+  });
+}
+
+export function updateElement(parentElement, newNode, oldNode, index = 0) {
+  // 새로운 노드가 추가될 경우
+  if (newNode && !oldNode) {
+    parentElement.appendChild(createElement(newNode));
+    return;
+  }
+
+  // 이전 노드가 삭제된 경우
+  if (!newNode && oldNode) {
+    parentElement.removeChild(parentElement.childNodes[index]);
+    return;
+  }
+
+  // 노드의 타입이 다른경우
+  if (newNode.type !== oldNode.type) {
+    parentElement.replaceChild(
+      createElement(newNode),
+      parentElement.childNodes[index],
+    );
+    return;
+  }
+
+  // 텍스트 노드일 경우
+  if (
+    typeof newNode === "string" &&
+    typeof oldNode === "string" &&
+    newNode !== oldNode
+  ) {
+    parentElement.replaceChild(
+      createElement(newNode),
+      parentElement.childNodes[index],
+    );
+    return;
+  }
+
+  if (isChangedAttributes(newNode.props, oldNode.props)) {
+    updateAttributes(
+      parentElement.childNodes[index],
+      newNode.props,
+      oldNode.props,
+    );
+  }
+
+  const newNodeChildren = newNode.children ?? [];
+  const oldNodeChildren = oldNode.children ?? [];
+  const maxChildren = Math.max(newNodeChildren.length, oldNodeChildren.length);
+
+  for (let i = 0; i < maxChildren; i++) {
+    updateElement(
+      parentElement.childNodes[index],
+      newNodeChildren[i],
+      oldNodeChildren[i],
+      i,
+    );
+  }
+}
```

Comment:
이 함수 다 좋네요 bb 그런데 중복되는 로직이 있어보이고, 텍스트 노드 업데이트 로직을 textContent 변경으로 구현해보면 더 좋을것 같아서 코드를 제안해봅니다~!

```suggestion
export function updateElement(parentElement, newNode, oldNode, index = 0) {
  if (!oldNode && newNode) {
    parentElement.appendChild(createElement(newNode));
    return;
  }

  if (!newNode) {
    parentElement.removeChild(parentElement.childNodes[index]);
    return;
  }

  const childElement = parentElement.childNodes[index];

  if (newNode.type !== oldNode.type) {
    parentElement.replaceChild(createElement(newNode), childElement);
    return;
  }

  if (typeof newNode === "string" && newNode !== oldNode) {
    childElement.textContent = newNode; 
  }

  if (isChangedAttributes(newNode.props, oldNode.props)) {
    updateAttributes(childElement, newNode.props, oldNode.props);
  }

  const maxLength = Math.max(newNode.children?.length || 0, oldNode.children?.length || 0);
  for (let i = 0; i < maxLength; i++) {
    updateElement(childElement, newNode.children[i], oldNode.children[i], i);
  }
}
```